### PR TITLE
chore: improve eslint

### PR DESCRIPTION
**Pull request by septs**
closes #

### CODE REVIEW

**Review by guanbinrui — DISMISSED** [review]

File: packages/.eslintrc.json [hunk 1]
```diff
@@ -23,7 +23,18 @@
         "lodash/import-scope": ["error", "member"],
         "unused-imports/no-unused-imports-ts": "warn",
         "@typescript-eslint/await-thenable": "error",
+        "@typescript-eslint/consistent-indexed-object-style": ["off", "record"],
```
Comment (Jack-Works) [inline packages/.eslintrc.json:26]: Remove this rule if it turned off.

File: packages/.eslintrc.json [hunk 2]
```diff
@@ -23,7 +23,18 @@
...
         "@typescript-eslint/no-for-in-array": "error",
-        "@typescript-eslint/consistent-indexed-object-style": ["off", "record"]
+        "@typescript-eslint/no-implicit-any-catch": "error",
```
Comment (Jack-Works) [inline packages/.eslintrc.json:28]: Don't add this. We will have this in TypeScript itself in 4.4. 
See https://github.com/microsoft/TypeScript/pull/41013

File: packages/.eslintrc.json [hunk 3]
```diff
@@ -23,7 +23,18 @@
...
+        "@typescript-eslint/no-implied-eval": "error",
+        "@typescript-eslint/no-invalid-this": "error",
+        "@typescript-eslint/no-invalid-void-type": "error",
```
Comment (Jack-Works) [inline packages/.eslintrc.json:31]: I have a concern about this rule. https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-invalid-void-type.md

It bans the following code:

```ts
interface Interface {
  lambda: () => void;
}
```

Which I don't think it is necessary to do so

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/maskbook/src/plugins/External/index.ts [hunk 4]
```diff
@@ -5,7 +5,6 @@ registerPlugin({
     ...base,
     SNSAdaptor: {
         load: () => import('./SNSAdaptor'),
-        hotModuleReload: (hot) =>
-            module.hot && module.hot.accept('./SNSAdaptor/index', () => hot(import('./SNSAdaptor'))),
+        hotModuleReload: (hot) => module.hot?.accept('./SNSAdaptor/index', () => hot(import('./SNSAdaptor'))),
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/External/index.ts:8]:
```suggestion
        hotModuleReload: (hot) => import.meta.webpackHot?.accept('./SNSAdaptor/index', () => hot(import('./SNSAdaptor'))),
```

File: packages/.eslintrc.json (hunk 2 above)
Comment (septs) [reply packages/.eslintrc.json]: https://github.com/typescript-eslint/typescript-eslint/pull/3705

File: packages/maskbook/src/utils/promiEvent.ts [hunk 5]
```diff
@@ -128,9 +128,9 @@ export function iteratorToPromiEvent(
     },
```
Comment (guanbinrui) [inline packages/maskbook/src/utils/promiEvent.ts:128]: Please remove this file. We don't need it anymore.

File: packages/web3-shared/src/hooks/useMulticall.ts [hunk 6]
```diff
@@ -97,11 +89,7 @@ export function useMutlicallStateDecoded<
                     value: decodeOutputString(web3, outputs, raw) as R,
                 }
             } catch (error) {
-                return {
-                    raw,
-                    error: error as Error,
-                    value: null,
-                }
+                return { raw, error, value: null }
```
Comment (guanbinrui) [inline packages/web3-shared/src/hooks/useMulticall.ts:92]:
```typescript
if (error instanceof Error) {
    // -
}
```

File: packages/maskbook/src/social-network-adaptor/facebook.com/automation/pasteTextToComposition.ts [hunk 7]
```diff
@@ -48,8 +48,8 @@ async function openPostDialogFacebook() {
         await timeout(new MutationObserverWatcher(isMobileFacebook ? activated : dialog), 2000)
         console.log('Dialog appeared')
     } catch {}
-    function clickFailed(e: unknown) {
-        console.warn(e)
+    function clickFailed(error: unknown) {
+        console.warn(error)
         if (!dialog.evaluate()[0]) alert('请点击输入框')
```
Comment (guanbinrui) [inline packages/maskbook/src/social-network-adaptor/facebook.com/automation/pasteTextToComposition.ts:53]:
```suggestion
        if (!dialog.evaluate()[0]) alert('Click the input box, please.')
```

**Review by guanbinrui — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Review by zhouhanseng — APPROVED** [review]

**Review by Himself65 — APPROVED** [review]

File: packages/external-plugin-previewer/src/index.tsx [hunk 8]
```diff
@@ -16,7 +16,9 @@ export function MaskExternalPluginPreviewRenderer({ pluginBase, payload, script,
             instance.$mount({ target: dom })
             return () => instance.$destroy()
         } catch (e) {
-            onError?.(e)
+            if (e instanceof Error) {
```
Comment (Jack-Works) [inline packages/external-plugin-previewer/src/index.tsx:19]: This change actually changes the semantics and might break the code. Please always re-throw the error

File: packages/maskbook/src/extension/background-script/EthereumServices/nonce.ts [hunk 9]
```diff
@@ -30,7 +30,9 @@ class NonceManager {
                     this.lock()
                     callback(null, await getTransactionCount(this.address))
                 } catch (e) {
-                    callback(e)
+                    if (e instanceof Error) {
+                        callback(e)
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/background-script/EthereumServices/nonce.ts:34]: This change actually changes the semantics and might break the code. Please always re-throw the error

File: packages/maskbook/src/extension/background-script/EthereumServices/send.ts [hunk 10]
```diff
@@ -173,7 +173,9 @@ export async function INTERNAL_send(
                 break
         }
     } catch (error) {
-        callback(error)
+        if (error instanceof Error) {
+            callback(error)
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/background-script/EthereumServices/send.ts:177]: This change actually changes the semantics and might break the code. Please always re-throw the error

File: packages/maskbook/src/extension/background-script/StorageService.ts [hunk 11]
```diff
@@ -27,7 +27,9 @@ class MutexStorage<T extends browser.storage.StorageValue> {
                     const stored = await browser.storage.local.get(key)
                     callback(null, (stored ?? {})[key] as T)
                 } catch (e) {
-                    callback(e)
+                    if (e instanceof Error) {
+                        callback(e)
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/background-script/StorageService.ts:31]: This change actually changes the semantics and might break the code. Please always re-throw the error

File: packages/maskbook/src/extension/background-script/StorageService.ts [hunk 12]
```diff
@@ -48,7 +50,9 @@ class MutexStorage<T extends browser.storage.StorageValue> {
                     await browser.storage.local.set({ [key]: value })
                     callback(null)
                 } catch (e) {
-                    callback(e)
+                    if (e instanceof Error) {
+                        callback(e)
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/background-script/StorageService.ts:54]: This change actually changes the semantics and might break the code. Please always re-throw the error

File: packages/web3-shared/src/hooks/useMulticall.ts [hunk 13]
```diff
@@ -64,10 +54,12 @@ export function useMulticallCallback() {
                     results: returnData,
                 })
             } catch (error) {
-                setMulticallState({
-                    type: MulticalStateType.FAILED,
-                    error,
-                })
+                if (error instanceof Error) {
```
Comment (Jack-Works) [inline packages/web3-shared/src/hooks/useMulticall.ts:57]: This change actually changes the semantics and might break the code. Please always re-throw the error

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**